lxd/certificates.go

```diff
@@ -543,6 +583,48 @@ func certificatesPost(d *Daemon, r *http.Request) response.Response {
 		}
 
 		name = remoteHost
+	} else if req.Token {
+		address, err := node.HTTPSAddress(d.db)
+		if err != nil {
+			return response.InternalError(err)
+		}
+		addresses, err := util.ListenAddresses(address)
+		if err != nil {
+			return response.InternalError(err)
+		}
+
+		// Generate join secret for new client. This will be stored inside the join token operation and will be
+		// supplied by the joining client (encoded inside the join token) which will allow us to lookup the correct
+		// operation in order to validate the requested joining client name is correct and authorised.
+		joinSecret, err := shared.RandomCryptoString()
+		if err != nil {
+			return response.InternalError(err)
+		}
+
+		// Generate fingerprint of network certificate so joining member can automatically trust the correct
+		// certificate when it is presented during the join process.
+		fingerprint, err := shared.CertFingerprintStr(string(d.endpoints.NetworkPublicKey()))
+		if err != nil {
+			return response.InternalError(err)
+		}
+
+		if req.Projects == nil {
+			req.Projects = []string{}
+		}
+
+		meta := map[string]interface{}{
+			"secret":      joinSecret,
+			"fingerprint": fingerprint,
+			"addresses":   addresses,
```

Comment:
No need to put those two in the operation data

Reply:
These fields are in the operation data because the admin uses this data to construct the token. If these were missing, the admin would need to call `GetServer()` to get the addresses and fingerprint which IMHO is an unnecessary call.

Reply:
Odd but nevermind, let's keep it the same as cluster tokens then :)